File: src/009/p009.go
package main

import "fmt"

func solve() int {
	product := 0
	for a := 0; a < 500; a++ {
		for b := a + 1; b < 500; b++ {
			c := 1000 - a - b
			sum := a + b + c
			if a*a+b*b == c*c && sum == 1000 {
				product = a * b * c
			}
		}
	}
	return product
}

func main() {
	result := solve()
	fmt.Println(result)
}

File: src/009/p009.lua
function solve()
  local product = 1
  for a=0, 500 do
    for b=a+1, 500 do
      local c = 1000 - b - a
      local sum = a + b + c
      if a*a + b*b == c*c and sum == 1000 then
        product = a * b * c
      end
    end
  end
  return product
end

local result = solve()
print(result)
